public/js/index.js

```diff
@@ -1,5 +1,32 @@
-function displayMessage() {
```

Comment:
Yes. I just pasted a local code snippet.

Yu are right! karma is complaining because...
- It cannot find any `app` node in the DOM.
- It requires to load a valid product array during the startup.

I pushed a work around to fix the unit tests but definitely I need help with this part.